packages/react-dropdown/src/Categories.js

```diff
@@ -122,10 +184,20 @@ export default function DropdownCategories({
     <div>
       {categoriesedItems.map(category => {
         const categoryId = category.id;
+
+        const isMultipleCategorySelectionEnabled =
+          enableCategorySelection && multiselect;
+
         const isHighlighted =
           highlightedIndex != null &&
-          highlightedIndex >= category.firstIndex &&
-          highlightedIndex <= category.lastIndex;
+          ((typeof highlightedIndex === 'number' &&
```

Comment:
This is very hard to read, also, isn't `highlightedIndex` either a number, or a string? Shouldn't we check for both? (or for `!= null`)

Reply:
I tried to simplify it like this:
```
const isHighlighted =
          isCategoryItemHighlighted(
            highlightedIndex,
            category.firstIndex,
            category.lastIndex
          ) ||
          isCategoryHighlighted(
            isMultipleCategorySelectionEnabled,
            highlightedIndex,
            categoryId
          );
```